build/org.eclipse.cdt.build.gcc.core/src/org/eclipse/cdt/build/gcc/core/ClangToolChain.java

```diff
@@ -28,7 +32,29 @@ public class ClangToolChain extends GCCToolChain {
 
 	public ClangToolChain(IToolChainProvider provider, Path pathToToolChain, String arch,
 			IEnvironmentVariable[] envVars) {
-		super(provider, pathToToolChain, arch, envVars);
+		super(provider, pathToToolChain, arch, addClangEnvVars(envVars));
+	}
+
+	private static IEnvironmentVariable[] addClangEnvVars(IEnvironmentVariable[] envVars) {
+		List<IEnvironmentVariable> envVarsNew = new ArrayList<>(Arrays.asList(envVars));
+		/*
+		 * Set CC and CXX environment variables for clang and clang++. This is equivalent to setting these in the
+		 * CMakeLists.txt:
+		 * set(CMAKE_C_COMPILER clang)
+		 * set(CMAKE_CXX_COMPILER clang++)
+		 */
+		addIfAbsent("cc", "clang", envVarsNew); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
cc and cxx should be upper case!

Reply:
> Does envVarsNew already include the system environment,
No; its the env from the core build and/or any env vars set by the user for a "User Defined Toolchain".